src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -47,6 +50,22 @@ public static class ActivatorUtilities
                 throw new InvalidOperationException(SR.CannotCreateAbstractClasses);
             }
 
+            if (!s_constructorInfoEx.TryGetValue(instanceType, out ConstructorInfoEx[]? constructors))
+            {
+                ConstructorInfo[] ctors = instanceType.GetConstructors();
+                ConstructorInfoEx[] temp = new ConstructorInfoEx[ctors.Length];
+                for (int i = 0; i < ctors.Length; i++)
+                {
+                    temp[i] = new ConstructorInfoEx(ctors[i]);
+                }
+
+                // Overwrite if another thread already set; they would contain the same information.
+                s_constructorInfoEx[instanceType] = temp;
+
+                constructors = temp;
```

Comment:
Could also be:
```C#
constructors = s_constructorInfoEx.GetOrAdd(instanceType, temp);
```
and then everyone will have the same instance.

Reply:
Sure; should be rare.